aws-rds-customdbengineversion/src/main/java/software/amazon/rds/customdbengineversion/CreateHandler.java

```diff
@@ -0,0 +1,124 @@
+package software.amazon.rds.customdbengineversion;
+
+import java.util.HashSet;
+
+import com.amazonaws.util.StringUtils;
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.rds.common.handler.Commons;
+import software.amazon.rds.common.handler.HandlerConfig;
+import software.amazon.rds.common.handler.Tagging;
+import software.amazon.rds.common.util.IdentifierFactory;
+
+public class CreateHandler extends BaseHandlerStd {
+
+    private final static IdentifierFactory dbCustomEngineVersionIdentifierFactory = new IdentifierFactory(
+            STACK_NAME,
+            RESOURCE_IDENTIFIER,
+            RESOURCE_ID_MAX_LENGTH
+    );
+
+    public CreateHandler() {
+        this(CUSTOM_ENGINE_VERSION_HANDLER_CONFIG_10H);
+    }
+
+    public CreateHandler(HandlerConfig config) {
+        super(config);
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<RdsClient> proxyClient,
+            final Logger logger) {
+
+        final ResourceModel model = request.getDesiredResourceState();
+
+        assertCustomDbEngineVersionStableStatus(model.getStatus());
+
+        if (StringUtils.isNullOrEmpty(model.getEngineVersion())) {
```

Comment:
Does RDS Service deduces engine version if it was not provided explicitly? If it doesn't, we are hardcoding defaults which might be detrimental in the future. E.g. Oracle Custom starts supporting 21c, and CFN would still default to 19c because it was hard-coded in the handler.

I don't think this is something we want to do.